### PR TITLE
Integrate RN Nightly Build 8/12

### PR DESCRIPTION
**Pull request by chiaramooney**
### Type of Change
Integration

### What
Commits: https://github.com/facebook/react-native/compare/a21a1f845...23429330a

Notable Changes:
- Bump ESLint https://github.com/facebook/react-native/commit/9797388847cdf5338c3a2fc9a2d8856b920e76d7
- Bump Gradle https://github.com/facebook/react-native/commit/5c8186623ae15388949cfc4143edae86863a447b
- Update RN CLI https://github.com/facebook/react-native/commit/c504d038c470f7a13fb345f57261172c7c85248c
- Remove Forking from Invalid Macro Syntax https://github.com/facebook/react-native/commit/fc26dbfce0ed16edf1efcfebc01dd4ca584eb902
- Remove Forking from Invalid Attribute Syntax https://github.com/facebook/react-native/commit/33dbb6c40ccf44e4b4c879270b6286e2829d057b
- Upgrade Flow https://github.com/facebook/react-native/commit/39b48b1c1d0a45c23ee2a5c6daac4434ee2c84f6
- Bump Gradle https://github.com/facebook/react-native/commit/7a911e073094b533cb5a7ce76932c02f83f4fe5d

NOTE: With the RN CLI bump metro-config now must have rigid versioning, due to RN CLI dependency change for metro-config to rigid versioning as well. 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/10683)

### CODE REVIEW

File: packages/@react-native-windows/virtualized-list/src/VirtualizedList.js [hunk 1]
```diff
@@ -426,7 +176,7 @@ class VirtualizedList extends React.PureComponent<Props, State> {
     viewOffset?: number,
     viewPosition?: number,
     ...
-  }) {
+  }): $FlowFixMe {
```
Comment (chrisglein) [inline packages/@react-native-windows/virtualized-list/src/VirtualizedList.js:179]: When are these $FlowFixMe comments supposed to be addressed?

File: vnext/ReactCommon/TEMP_UntilReactCommonUpdate/react/renderer/core/LayoutableShadowNode.cpp [hunk 2]
```diff
@@ -151,8 +151,21 @@ LayoutMetrics LayoutableShadowNode::computeRelativeLayoutMetrics(
   }
 
   // ------------------------------
-
-  auto transformedFrames = calculateTransformedFrames(shadowNodeList, policy);
+  // TODO: T127619309 remove after validating that T127619309 is fixed
```
Comment (chrisglein) [inline vnext/ReactCommon/TEMP_UntilReactCommonUpdate/react/renderer/core/LayoutableShadowNode.cpp:154]: Is this our TODO, or is it from the integrate?

File: vnext/ReactCommon/TEMP_UntilReactCommonUpdate/react/renderer/mapbuffer/MapBufferBuilder.cpp [hunk 3]
```diff
@@ -0,0 +1,182 @@
+/*
```
Comment (chrisglein) [inline vnext/ReactCommon/TEMP_UntilReactCommonUpdate/react/renderer/mapbuffer/MapBufferBuilder.cpp:1]: What's with this TEMP_UntilReactCommonUpdate folder?

File: packages/@react-native-windows/virtualized-list/src/VirtualizedList.js (hunk 1 above)
Comment (acoates-ms) [reply packages/@react-native-windows/virtualized-list/src/VirtualizedList.js:179]: Generally meta's process for updating the version of flow is an automated process that will make changes to the code to cast things as `$FlowFixMe` which amounts to someting similar to `any` in TS.  That way the flow update is unblocked.  Then they can go through later and provide better types.  I assume this change came from meta.

File: vnext/ReactCommon/TEMP_UntilReactCommonUpdate/react/renderer/core/LayoutableShadowNode.cpp (hunk 2 above)
Comment (acoates-ms) [reply vnext/ReactCommon/TEMP_UntilReactCommonUpdate/react/renderer/core/LayoutableShadowNode.cpp:154]: T127619309 looks like a meta task reference, so I assume its their todo.

File: vnext/ReactCommon/TEMP_UntilReactCommonUpdate/react/renderer/mapbuffer/MapBufferBuilder.cpp (hunk 3 above)
Comment (acoates-ms) [reply vnext/ReactCommon/TEMP_UntilReactCommonUpdate/react/renderer/mapbuffer/MapBufferBuilder.cpp:1]: Yeah, if we need to fork files in core can we do a better job of putting comments in the forked file about what we changed.  -- I would also expect to see an entry in overrides.json for this file -- preferably with a reference to an issue tracking removing the fork.

**Review by TatianaKapos — APPROVED** [review]

**Pull request merged**